### ISSUE TITLE
support for parameter binding

### ISSUE THREAD

**Issue by ananthakumaran** [mneudert/instream#49]
Influxdb provides an option to specify [bind parameters](https://docs.influxdata.com/influxdb/v1.7/tools/api/#bind-parameters). This would come in handy when one wants to avoid query injection. Currently, the query api doesn't expose this functionality.

**Comment by mneudert** [comment]
Sounds like a good idea! Would the following way to pass parameters be reasonably understandable?

```elixir
MyApp.MyConnection.query(
  "SELECT [...] WHERE [key] = $<placeholder>",
  params: %{
    "<placeholder>" => <placeholder_value>
  }
)
```

Basically speaking the placeholders could be passed as a map with binary keys and then sent off to InfluxDB as passed.

**Comment by ananthakumaran** [comment]
Yes, looks perfect.

**Comment by mneudert** [comment]
Parameter binding should now be available in the just released version `0.21.0`. The parameters passed in the map should be fine as both binaries and atoms.

Please open an issue or comment here if you think anything is missing or not working as intended :)

---

According to the [InfluxDB source](https://github.com/influxdata/influxdb/blob/v1.7.6/services/httpd/handler.go#L462) the parameters are expected to be passed as a JSON document. This may require you to now configure `:json_encoder` if you have also configured `:json_decoder`.

If you encounter any problems with this conversion please comment here or open an issue :)
